orca-remote-stage/src/main/kotlin/com/netflix/spinnaker/orca/remote/stage/messages/RemoteStageStatusMessageHandler.kt

```diff
@@ -0,0 +1,44 @@
+package com.netflix.spinnaker.orca.remote.stage.messages
+
+import com.netflix.spinnaker.orca.api.pipeline.models.ExecutionType
+import com.netflix.spinnaker.orca.api.pipeline.remote.messages.RemoteStageMessage
+import com.netflix.spinnaker.orca.api.pipeline.remote.messages.RemoteStageStatusMessage
+import com.netflix.spinnaker.orca.pipeline.persistence.ExecutionRepository
+import com.netflix.spinnaker.orca.remote.stage.StageExecutionNotFound
+import org.slf4j.LoggerFactory
+import org.springframework.stereotype.Component
+
+@Component
+class RemoteStageStatusMessageHandler(
+  val executionRepository: ExecutionRepository
+) : RemoteStageMessageHandler<RemoteStageStatusMessage> {
+
+  private val log by lazy { LoggerFactory.getLogger(javaClass) }
+
+  override fun supports(message: Class<RemoteStageMessage>): Boolean {
+    return message == RemoteStageStatusMessage::class.java
+  }
+
+  override fun handle(message: RemoteStageStatusMessage) {
+    log.info("Remote stage execution '{}' status received with '{}'", message.stageExecutionId, message.status.toString())
+    val pipelineExecution = executionRepository.retrieve(ExecutionType.PIPELINE, message.pipelineExecutionId)
+    val stageExecution = pipelineExecution.stages.find { it.id == message.stageExecutionId }
+
+    if (stageExecution != null) {
+      stageExecution.status = message.status
+      stageExecution.outputs = message.outputs
+      stageExecution.context["remoteTasks"] = message.tasks
+      stageExecution.context["remoteStartTime"] = message.startTime
+      stageExecution.context["remoteEndTime"] = message.endTime
+      stageExecution.context["remoteError"] = message.error
```

Comment:
Similarly here, I'm confused why an error in the remote stage would not impact the stage inside of Orca?

Reply:
Don't we have fairly standardized handling for errors, is there a need to go off the path here?

If we go this messages on queue route, `outputs` could be quite large. Are messages intended to contain context or just pointers to context?

Reply:
Also question whether we should align the way these updates get applied with [TaskController. updatePipelineStage()](https://github.com/spinnaker/orca/blob/master/orca-web/src/main/groovy/com/netflix/spinnaker/orca/controllers/TaskController.groovy#L474).